packages/client/hmi-client/src/components/code/tera-code-dynamic.vue

```diff
@@ -0,0 +1,81 @@
+<template>
+	<h3>{{ filename ?? '' }}</h3>
+	<Card
+		v-for="(block, index) in codefile?.dynamics?.block"
+		class="code-dynamic-card-container"
+		:key="index"
+	>
+		<template #title>
+			Code Block
+			<div v-if="!editable">
+				<!-- <Button text icon="pi pi-pencil" @click="editable = true" /> -->
+				<Button text icon="pi pi-trash" @click="deleteCodeBlock(index)" />
+			</div>
+		</template>
+		<template #content>
+			<div>
+				Lines {{ extractDynamicRows(block).startRow + 1 }} to
+				{{ extractDynamicRows(block).endRow + 1 }}
+			</div>
+		</template>
```

Comment:
We are missing the `description` field

Reply:
I think we may need to change how we store the dynamics in TDS in order to show the name and description?  The issue is that each file can have multiple code blocks but each code block within a file will all have the same name and description:
```
"chime-1.1.5/.bldr.jl": {
            "dynamics": {
                "name": "name",
                "description": "desc",
                "block": [
                    "L11-L20",
                    "L1-L2"
                ]
            }
        },  
```

Instead maybe it should be something like this?

```
"chime-1.1.5/.bldr.jl": {
            "dynamics": [
              {
                "name": "name1",
                "description": "desc1",
                "block": "L11-L20"
               },
               {
                "name": "name2",
                "description": "desc2",
                "block": "L11-L20"
                    "L1-L2"
               }
            ]
        },  
```

Reply:
Let's talk about it. Because this is not how I remember defining code block with Brandon on the `data-service`.

Reply:
Ill post this comment in the eng channel